packages/blockchain-wallet-v4-frontend/src/layouts/Dex/DexHeader/DexHeader.tsx

```diff
@@ -1,103 +1,230 @@
+      })
+    },
+    {
+      key: 'prices',
+      label: formatMessage({
+        defaultMessage: 'Prices',
+        id: 'navbar.primary.prices'
+      })
+    },
+    {
+      dot: true,
+      key: 'earn',
+      label: formatMessage({
+        defaultMessage: 'Earn',
+        id: 'navbar.primary.earn'
+      })
+    }
+  ]
 
-const mapStateToProps = (state) => ({
-  ethBalanceR: selectors.balances.getCoinNonCustodialBalance('ETH')(state)
-})
+  if (isNftExplorerEnabled) {
+    navigationTabs.push({
+      dot: true,
+      key: 'nfts',
+      label: formatMessage({
+        defaultMessage: 'NFTs',
+        id: 'navbar.primary.nfts'
+      })
+    })
+  }
 
-const connector = connect(mapStateToProps)
-type Props = ConnectedProps<typeof connector> & OwnProps
+  if (isDexEnabled) {
+    navigationTabs.push({
+      dot: true,
+      key: 'dex',
+      label: formatMessage({
+        defaultMessage: 'DEX',
+        id: 'navbar.primary.dex'
+      })
+    })
+  }
 
-export default connector(DexHeader)
+  const accountDropdownItems = [
+    {
+      key: 'general',
+      label: formatMessage({
+        defaultMessage: 'General',
+        id: 'navbar.dropdown.general'
+      })
+    },
+    {
+      key: 'security',
+      label: formatMessage({
+        defaultMessage: 'Security',
+        id: 'navbar.dropdown.security'
+      })
+    },
+    {
+      key: 'trading-limits',
+      label: formatMessage({
+        defaultMessage: 'Trading Limits',
+        id: 'navbar.dropdown.tradingLimits'
+      })
+    },
+    {
+      key: 'preferences',
+      label: formatMessage({
+        defaultMessage: 'Preferences',
+        id: 'navbar.dropdown.preferences'
+      })
+    },
+    {
+      key: 'wallets-addresses',
+      label: formatMessage({
+        defaultMessage: 'Wallets & Addresses',
+        id: 'navbar.dropdown.walletsAndAddresses'
+      })
+    },
+    {
+      key: 'refer-friend',
+      label: formatMessage({
+        defaultMessage: 'Refer a friend',
+        id: 'navbar.dropdown.referFriend'
+      })
+    },
+    {
+      key: 'tax-center',
+      label: formatMessage({
+        defaultMessage: 'Tax Center',
+        id: 'navbar.dropdown.taxCenter'
+      })
+    }
+  ]
+
+  const iconActions = [
+    {
+      icon: () => <IconRefresh />,
+      label: formatMessage({
+        defaultMessage: 'Refresh',
+        id: 'navbar.icons.refresh'
+      }),
+      onClick: () => dispatch(actions.components.refresh.refreshClicked())
+    },
+    {
+      icon: () => <IconPresent />,
+      label: formatMessage({
+        defaultMessage: 'Refer',
+        id: 'navbar.icons.refer'
+      }),
+      // TODO: Same as click on "Refer a Friend" from a dropdown
+      onClick: () => undefined
+    }
+  ]
+
+  return (
+    <Navigation
+      defaultSelected={selectedTab}
+      iconActions={iconActions}
+      navigationTabs={navigationTabs}
+      dropdownSecondSectionItems={accountDropdownItems}
+      onSelectedChange={(key) => {
```

Comment:
I think this is pretty big for being inside as a prop, wouldn't it be better to have it outside?

In a pretty minor improvement, you could do 
```
case 'home':
case 'prices':
case 'earn':
case 'nfts':
case 'dex':
  dispatch(actions.router.push('/' + key))
  break
```